From fefe8133c22adbf9ef3e924c8d57fc2858d63886 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Sun, 20 Nov 2022 11:54:55 +0200
Subject: [PATCH] [SW-66410] gaudi3: init psoc drain

Psoc drain is enabled by default in HW, so all there is left to do
is adjusting the timeout configuration.

Change-Id: I365b5b7968e3b4a87cbf32639880ed47851e759c
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index de8fa44..a0ec387 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -212,8 +212,20 @@
 	regmap_writel(val, mmD0_PCIE_WRAP_LBW_DRAIN_CFG);
 }
 
+static void hl_psoc_drain_init(void)
+{
+	/* If FW skip enabled return */
+	if (hl_skip_fw_cfg())
+		return;
+
+	regmap_writel(HBW_DRAIN_TIMEOUT, mmD0_PARC_GLOBAL_CONF_MST_HBW_DRAIN_TIMEOUT);
+	regmap_writel(LBW_DRAIN_TIMEOUT, mmD0_PARC_GLOBAL_CONF_MST_LBW_DRAIN_TIMEOUT);
+	/* psoc drain is enabled in HW by default */
+}
+
 void hl_plat_early_init(void)
 {
+	hl_psoc_drain_init();
 	hl_lbw_early_write_resp_enable();
 	init_prstn();
 	hl_dev_cpu_pll_init();
